common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java

```diff
@@ -105,11 +106,28 @@
     public BrokerRequest brokerRequestFromAcquireTokenParameters(@NonNull final InteractiveTokenCommandParameters parameters) {
         Logger.info(TAG, "Constructing result bundle from AcquireTokenOperationParameters.");
 
-        final String extraQueryStringParameter = parameters.getExtraQueryStringParameters() != null ?
-                QueryParamsAdapter._toJson(parameters.getExtraQueryStringParameters())
+        final List<Pair<String, String>> queryParams = parameters.getExtraQueryStringParameters();
+        final List<Pair<String, String>> extraQueryStringParameters = queryParams != null ?
+                new ArrayList<>(queryParams) : new ArrayList<Pair<String, String>>();
+
+        final AzureActiveDirectorySlice sliceData = parameters.getAuthority().getSlice();
+        if (sliceData != null) {
+            String slice = sliceData.getSlice();
+            if (slice != null) {
+                extraQueryStringParameters.add(new Pair(AzureActiveDirectorySlice.SLICE_PARAMETER, slice));
+            }
+            String dc = sliceData.getDC();
+            if (dc != null) {
+                extraQueryStringParameters.add(new Pair(AzureActiveDirectorySlice.DC_PARAMETER, dc));
```

Comment:
>extraQueryStringParameters [](http://example.com/codeflow?start=16&length=26)

Should we add a UT that verifies that the final request contains these query params..

Reply:
This I don't care about here.  I'm writing a UT that the translated InteractiveTokenRequestParamaters contains the relevant values when round-tripped through this code, and that's quite involved enough.